### DISCUSSION TITLE
Lose checked item in ST table when I navigate using pagination

### DISCUSSION THREAD

**Discussion by mouad-mahfoud** [ng-alain/ng-alain#2371] [Q&A]
Hello,
 
i have this component 

`import { Component } from '@angular/core';

import { STChange, STColumn, STData } from '@delon/abc/st';

@Component({
  selector: 'components-st-checkbox',
  template: ` <div class="mb-md">
      <button nz-button (click)="st.checkAll(true)">All</button>
      <button nz-button (click)="st.clearCheck()">Clean</button>
    </div>
    <st
      #st
      [data]="data"
      [columns]="columns"
      (change)="change($event)"
      [ps]="1"
    ></st>`,
})
export class ComponentsStCheckboxComponent {
  data = [
    {
      id: 13,
      creeLe: '2023-02-07T11:28:41.915+00:00',
      creePar: 'assureur@qualitadd.com',
      modifieLe: '2023-02-07T11:28:41.924+00:00',
      modifiePar: 'assureur@qualitadd.com',
      identifiant: 'TRS-3175008',
      libelle: 'T3',
      description: 'T3',
      criticiteTierId: 2,
      statut: 2,
      messageAcceuil:
        'Cher délégataire, nous avons bien pris en compte, dans le questionnaire de cette année, que l’activité d’infogérance a été renouvelée avec vous sur ces 4 prochaines années.',
      pieceJointes: [],
    },
    {
      id: 4,
      creeLe: '2023-02-02T11:24:21.131+00:00',
      creePar: 'assureur@qualitadd.com',
      modifieLe: '2023-02-02T11:24:21.176+00:00',
      modifiePar: 'assureur@qualitadd.com',
      identifiant: 'TRS-4370549',
      libelle: 'T1',
      description: 'TT11',
      criticiteTierId: 2,
      statut: 2,
      messageAcceuil:
        'Cher délégataire, nous avons bien pris en compte, dans le questionnaire de cette année, que l’activité d’infogérance a été renouvelée avec vous sur ces 4 prochaines années.',
      pieceJointes: [],
    },
    {
      id: 12,
      creeLe: '2023-02-07T11:28:26.374+00:00',
      creePar: 'assureur@qualitadd.com',
      modifieLe: '2023-02-07T11:28:26.385+00:00',
      modifiePar: 'assureur@qualitadd.com',
      identifiant: 'TRS-6230723',
      libelle: 'T2',
      description: 'T2',
      criticiteTierId: 1,
      statut: 2,
      messageAcceuil:
        'Cher délégataire, nous avons bien pris en compte, dans le questionnaire de cette année, que l’activité d’infogérance a été renouvelée avec vous sur ces 4 prochaines années.',
      pieceJointes: [],
    },
  ];
  columns: STColumn[] = [
    { title: '', index: 'identifiant', type: 'checkbox' },
    { title: 'Identifiant', index: 'identifiant', filter: { type: 'keyword' } },
    { title: 'Libellé', index: 'libelle', filter: { type: 'keyword' } },
    {
      title: 'Statut',
      index: 'statut',
      filter: {
        multiple: false,
        menus: [
          { text: 'Actif', value: 1 },
          { text: 'Non-Actif', value: 2 },
        ],
      },
    },
    {
      title: 'Criticité',
      index: 'criticiteTierId',
      filter: {
        multiple: true,
        menus: [
          { text: 'standard', value: 2 },
          { text: 'critique', value: 1 },
          { text: 'important', value: 3 },
        ],
      },
    },
  ];

  selectesRows = [];

  change(e: STChange): void {
    console.log('change', e);
  }
}`

I select an element in the first page and when I switch to the second page, I lose my selected element in the first page.

**Comment by Abdelhak-elgo** [comment]
hello , 

I've been working with a similar component, and I've encountered a similar issue. When I select an element in the first page and then switch to the second page, I lose my selected element in the first page.

To give you some context, I have attempted to implement a feature that ensures objects remain checked when users navigate between pages using the StTable component. To achieve this, I have created a function called dataProcess which should maintain the checked state of objects based on their IDs. Here is the relevant code snippet:

`dataProcess(data: STData[]): STData[] {
  return data.map((i, index) => {
    let id = this.familleDTO.sousFamilleSetDTO.map(id => id.idSousFamille);
    for (let y of id) {
      if (index === this.idST.indexOf(y)) {
        i.checked = true;
        return i;
      }
    }
    return i;
  });
}
`

In this code, dataProcess is designed to check objects by comparing them with their IDs and then get the object index in the StTable to marquee it as checked.

However, the issue I'm facing is that the checked state is not being preserved when navigating between pages using StTable. It seems that adding `[res]="{ process: dataProcess }"  `  in the St component

I would greatly appreciate any assistance or guidance in resolving this issue. If there are any suggestions or best practices for maintaining the checked state of objects when navigating between pages with StTable, please share them with me.

**Reply by cipchk** [reply]
SPA data recovery can take advantage of the service data sharing feature, or `RouteReuseStrategy` routing switching data without loss.
